activerecord/test/cases/reaper_test.rb

```diff
@@ -128,22 +128,37 @@ def test_connection_pool_starts_reaper_in_fork
           pool = ConnectionPool.new(pool_config)
           pool.checkout
 
+          # We currently have a bug somewhere which leads for this test case to be deadlocked
+          # and timeout after 30 minutes on the CI. Until that bug is fixed, this test is made
+          # to timeout after a short period of time to reduce the damage.
+          reader, writer = IO.pipe
+
           pid = fork do
+            reader.close
             pool = ConnectionPool.new(pool_config)
 
             conn, child = new_conn_in_thread(pool)
             child.terminate
 
             wait_for_conn_idle(conn)
+            writer.close
             if conn.in_use?
               exit!(1)
             else
               exit!(0)
             end
           end
 
-          Process.waitpid(pid)
-          assert_predicate $?, :success?
+          writer.close
+          completed = reader.wait_readable(20)
+          reader.close
+          if completed
+            Process.waitpid(pid)
+            assert_predicate $?, :success?
+          else
+            Process.kill("KILL", pid)
```

Comment:
You should call `Process.wait` even after a `SIGKILL` otherwise the child will linger around as a zombie.